packages/shared/src/pluginStore.ts

```diff
@@ -5,7 +5,72 @@ import type {
   RsbuildPluginAPI,
   RsbuildPlugins,
 } from './types';
-import { color, isFunction } from './utils';
+import { color } from './utils';
+
+class PluginTypeError extends Error {
+  constructor(plugin: unknown) {
+    super(
+      `Expect Rsbuild plugin instance to be an object, but got ${typeof plugin}.`,
+    );
+  }
+}
+
+class PluginMissingNameError extends Error {
+  constructor() {
+    super(`Rsbuild plugin should have a "name" field for identification.`);
+  }
+}
+
+class WebpackPluginLikeError extends Error {
+  constructor(plugin: object) {
+    const name = plugin.constructor?.name ?? 'SomeWebpackPlugin';
+    const message =
+      `${color.yellow(name)} looks like a webpack or Rspack plugin, ` +
+      `please use ${color.yellow('`tools.rspack`')} to register it:\n`;
+    const example = [
+      '  // rsbuild.config.ts',
+      '  export default {',
+      '    tools: {',
+      '      rspack: {',
+      `        plugins: [new ${name}()]`,
```

Comment:
No name here, you should test it.

![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/6cc996e0-1750-41d0-bcc4-b21a76a3ecf0)

Reply:
It seems to be the result of unexpected merging of https://github.com/web-infra-dev/rsbuild/pull/1156. This PR is intended to fix it.